### PR TITLE
Long-running form submit spinners

### PR DESCRIPTION
**Pull request by davecranwell**
First pass at addressing https://github.com/torchbox/wagtail/issues/305

910e1569e1eb28b54d385e3077b9dae9

75265d3b273d440d0d1ef22ddcdaaecc

Still to apply this to more forms

### CODE REVIEW

File: wagtail/wagtailadmin/static/wagtailadmin/js/core.js [hunk 1]
```diff
@@ -70,7 +70,8 @@ $(function(){
     /* close all dropdowns on body clicks */
     $(document).on('click', function(e){
         var relTarg = e.relatedTarget || e.toElement;
-        if(!$(relTarg).hasClass('dropdown-toggle')){
+        if(!$(relTarg).hasClass('dropdown-toggle') && !$(relTarg).closest('.dropdown').length){
+            console.log($(relTarget).closest('.dropdown'));
```
Comment (davecranwell) [inline wagtail/wagtailadmin/static/wagtailadmin/js/core.js]: Will remove

**Comment by tomdyson** [comment]
Looks good!

**Comment by tomdyson** [comment]
@joshbarr do you have time to review this?

**Comment by davecranwell** [comment]
- [ ] media embedding modal badly needs this spinner. The round trip to embed.ly seems to take forever.

**Comment by JoshBarr** [comment]
@tomdyson  sure thing, I can take a look at the media embedding modal. Are there any others you can think of while I'm there?

**Comment by JoshBarr** [comment]
Hey guys,

I was wondering whether we need to use setTimeout to disable the button? I think we can get the parent `form` from the current node and trigger submit from the click handler. Something like: 

```
// Trigger form submission if it's a submit button
if (this.form && this.getAttribute('type') === 'submit') {
    $(this.form).trigger('submit');
}

// Then carry on and disable the button + bootup the timeout to clear the spinner... 
```

Also, I've found an issue with trying to embed multiple media assets into the rich text field (in Chrome at least). If the first element is an embed (no other text nodes at the start of the block), trying to insert a second embed generates an `Uncaught HierarchyRequestError: Failed to execute 'insertBefore' on 'Node': Only one element on document allowed.` in `rangy-core.js:29`. I'll take a look at at one next.

**Comment by davecranwell** [comment]
Cheers for reviewing, @JoshBarr. As far as submitting the form goes, I don't have any issue with that. I was trying to avoid having to refer to forms at all purely to keep it all within the scope of the button, but if it's more reliable that's fine. I was hoping to avoid having to test the button type too, allowing this to be used for ajax forms, synchronous ones, and ones where true "submit" buttons aren't used.

**Comment by davecranwell** [comment]
Ready for merge, I believe.

**Comment by gasman** [comment]
There's a problem with the form submit logic from https://github.com/torchbox/wagtail/pull/1075#issuecomment-89169810 - the form submission won't include the name/value of the submit button, as it would when the submit action is triggered directly from the button. We use this to differentiate between the save/publish/submit actions on the page editor - consequently, with this patch applied, the 'publish' button fails to publish.

**Comment by davecranwell** [comment]
Ah yes, that was the main reason I didn't use `form.submit()` before, now I remember.

**Comment by davecranwell** [comment]
Reverted to version that supports passing of button names.

**Pull request merged**